07-Template-Distribution-Protocol.md

```diff
@@ -12,17 +12,18 @@ Template Providers MUST attempt to broadcast blocks which are mined using work t
 
 ## 7.1 `CoinbaseOutputDataSize` (Client -> Server)
 
-Ultimately, the pool is responsible for adding coinbase transaction outputs for payouts and other uses, and thus the Template Provider will need to consider this additional block size when selecting transactions for inclusion in a block (to not create an invalid, oversized block).
-Thus, this message is used to indicate that some additional space in the block/coinbase transaction be reserved for the pool’s use (while always assuming the pool will use the entirety of available coinbase space).
+Ultimately, the pool is responsible for adding coinbase transaction outputs for payouts and other uses, and thus the Template Provider will need to consider this additional block size, and sigops when selecting transactions for inclusion in a block (to not create an invalid, oversized block).
+Thus, this message is used to indicate that some additional space and sigops in the block/coinbase transaction be reserved for the pool’s use (while always assuming the pool will use the entirety of available coinbase space and sigops).
 
-The Job Declarator MUST discover the maximum serialized size of the additional outputs which will be added by the pool(s) it intends to use this work.
-It then MUST communicate the maximum such size to the Template Provider via this message.
-The Template Provider MUST NOT provide `NewMiningJob` messages which would represent consensus-invalid blocks once this additional size — along with a maximally-sized (100 byte) coinbase field — is added.
+The Job Declarator MUST discover the maximum serialized size of the additional outputs and sigops which will be added by the pool(s) it intends to use this work.
+It then MUST communicate it to the Template Provider via this message.
+The Template Provider MUST NOT provide `NewMiningJob` messages which would represent consensus-invalid blocks once this additional size and sigops — along with a maximally-sized (100 byte) coinbase script field — is added.
 Further, the Template Provider MUST consider the maximum additional bytes required in the output count variable-length integer in the coinbase transaction when complying with the size limits.
 
 | Field Name                          | Data Type | Description                                                                                     |
 | ----------------------------------- | --------- | ----------------------------------------------------------------------------------------------- |
 | coinbase_output_max_additional_size | U32       | The maximum additional serialized bytes which the pool will add in coinbase transaction outputs |
+| coinbase_output_max_sigops          | U16       | The maximum additional sigops which the pool will add in coinbase transaction outputs           |
```

Comment:
Let's add a comment in the text that this implies a limit of `65535` - slightly less than the consensus maximum of `80000`. This seems fine since I can't think of a (sane) use case of putting that many sigops in the coinbase.

Reply:
(this is done)